cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1355,7 +1355,7 @@ class device_row_comparator {
   __device__ constexpr bool operator()(size_type const lhs_index,
                                        size_type const rhs_index) const noexcept
   {
-    auto equal_elements = [=](column_device_view l, column_device_view r) {
+    auto equal_elements = [lhs_index, rhs_index, this](column_device_view l, column_device_view r) {
```

Comment:
These kind of changes are due to deprecation of implicitly captured `this` by value.
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0806r2.html
But, once I added `this` to the capture list, got another error that capturing this is redundant with `[=]` so I ended up specifying each variable to capture explicitly. This is the preferable way to do this, so no hard done.
It does seem like an ill-planned or implemented deprecation.